### DISCUSSION TITLE
Automate adding an asset to a release

### DISCUSSION THREAD

**Discussion by daveTheOldCoder** [community/community#78998] [Actions] [answered]
### Select Topic Area

Question

### Body

I have a workflow .yml script that creates a .zip. It's triggered by creating a new release. The .zip shows up as an artifact on the Action page of the repository.

Is it possible for the .yml script to place the .zip on the release page under Assets, along with the .zip and .tar.gz that GitHub generates?

I know that you can manually drop files there, but I'd like to automate it.

I looked through the documentation, and experimented, but couldn't figure out the file path for moving the .zip there, or if that's even possible.

**Comment by MrNull1** [comment]
Is there anything that puts the file in the main branch or is it automatic?

**Reply by daveTheOldCoder** [reply]
I don't know what "puts the file in the main branch" means.

The file is automatically placed on the Action page, as a release artifact, as I stated.

**Reply by MrNull1** [reply]
Then I have no idea 😅

**Reply by MrNull1** [reply]
I asked chatgpt cuz i've got notthing better to do and this is what it told me:

To automatically add the .zip file generated by your workflow as an asset to a GitHub release, you can use the GitHub Release Action in your workflow. Here's an example of how you can set this up in your workflow .yml file:

`
name: Create Release

on:
  release:
    types: [created]

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout code
        uses: actions/checkout@v2

      # Add your build and other necessary steps here

      - name: Create Release
        id: create_release
        uses: actions/create-release@v1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          tag_name: ${{ github.event.release.tag_name }}
          release_name: Release ${{ github.event.release.tag_name }}
          body: |
            Description of the release
          draft: false
          prerelease: false

      - name: Upload Release Asset
        id: upload-release-asset
        uses: actions/upload-release-asset@v1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          upload_url: ${{ steps.create_release.outputs.upload_url }}
          asset_path: ./path/to/your/zip/file.zip
          asset_name: your-zip-file.zip
          asset_content_type: application/zip
`
Replace ./path/to/your/zip/file.zip with the actual path to the .zip file generated by your workflow.

In this example, the actions/create-release@v1 action is used to create a new release, and the actions/upload-release-asset@v1 action is used to upload the .zip file as an asset to the release.

Make sure to replace GITHUB_TOKEN with a personal access token or use the default secrets.GITHUB_TOKEN if your repository is private.

With this setup, whenever a new release is created, the .zip file will be automatically added as an asset to the release.

**Comment by daveTheOldCoder** [comment]
Actually, I asked ChatGPT about the problem a few days ago, and got a similar answer. When I ran the script it provided, GitHub gave me errors about the tags after "with:" at the end.

`Unexpected input(s) 'upload_url', 'asset_path', 'asset_name', 'asset_content_type', valid inputs are ['repository', 'ref', 'token', 'ssh-key', 'ssh-known-hosts', 'ssh-strict', 'persist-credentials', 'path', 'clean', 'sparse-checkout', 'sparse-checkout-cone-mode', 'fetch-depth', 'fetch-tags', 'lfs', 'submodules', 'set-safe-directory', 'github-server-url']`

I've been looking at https://github.com/actions/upload-release-asset to see if I can use it.

**Reply by MrNull1** [reply]
I wonder if you are supposed to replace `'upload_url', 'asset_path', 'asset_name', 'asset_content_type'` with the tags `valid inputs are ['repository', 'ref', 'token', 'ssh-key', 'ssh-known-hosts', 'ssh-strict', 'persist-credentials', 'path', 'clean', 'sparse-checkout', 'sparse-checkout-cone-mode', 'fetch-depth', 'fetch-tags', 'lfs', 'submodules', 'set-safe-directory', 'github-server-url']`

**Reply by MrNull1** [reply]
In addition I could try to ask claude or another model xD

**Reply by daveTheOldCoder** [reply]
Replace them with what?

ChatGPT sometimes makes things up. I asked it how to do something with a .gitattributes file, and it told me to use a "zipexport" attribute, which apparently doesn't exist. It apologized after I pointed that out.

**Reply by MrNull1** [reply]
What I was saying was that in the error it told you about valid inputs, I wonder if it is telling you to use those instead. 
Sorry, I don't know a whole lot about GitHub.

**Comment by spenserblack** [comment, marked as answer]
Yeah, there are plenty of community-written actions to help you upload assets to a release. I use `softprops/action-gh-release`, but there might be better ones out there.

Here's an example:
https://github.com/spenserblack/gengo/blob/272e794ff895149152f942a9b71ceb39711a4fb9/.github/workflows/release.yml#L48-L56

**Reply by daveTheOldCoder** [reply]
Thanks. I was able to solve my problem using that example.

**Comment by berlimaloim** [comment]
como inisia

**Reply by daveTheOldCoder** [reply]
What does that mean?
